### PR TITLE
fix: combine event data in str or dict

### PR DESCRIPTION
**Pull request by Ernst79**
Combining `event_type` and `event_subtype` into `event_data`. 

`event_data` can be a string (e.g. `"toggle"`) or a dictionary (e.g. `{"rotate left": 3}`)

### CODE REVIEW

File: src/sensor_state_data/data.py [hunk 1]
```diff
@@ -289,8 +289,7 @@ def update_sensor(
     def fire_event(
         self,
         key: str,
-        event_type: str,
-        event_subtype: str | None = None,
+        event_data: str | dict[str, str | int | float],
```
Comment (bdraco) [inline src/sensor_state_data/data.py:292]: Is there any reason to not always make it a `dict`?

Comment (Ernst79) [reply src/sensor_state_data/data.py:292]: We can do that, one has to use `{"toggle": None}` if he only wants to send "toggle".

Comment (bdraco) [reply src/sensor_state_data/data.py:292]: I think thats fine. It should keep the consumer code a big simpler to not have to handle both cases

Comment (bdraco) [inline src/sensor_state_data/data.py:292]:
```suggestion
        event_data: dict[str, str | int | float | None],
```

File: src/sensor_state_data/value.py [hunk 2]
```diff
@@ -32,5 +32,4 @@ class BinarySensorValue(BaseValue):
 class Event(BaseValue):
     """A class that describes device events."""
 
-    event_type: str
-    event_subtype: str | None
+    event_data: str | dict[str, str | int | float]
```
Comment (bdraco) [inline src/sensor_state_data/value.py:35]:
```suggestion
    event_data: dict[str, str | int | float | None]
```

File: src/sensor_state_data/data.py (hunk 1 above)
Comment (bdraco) [reply src/sensor_state_data/data.py:292]: Another option might be to leave `event_type` as a `str` and then have `event_properties` be a `dict` so you have some level of normalization and the non-normalized data in the `dict`

Comment (Ernst79) [reply src/sensor_state_data/data.py]: If that is better, it is also fine with me. Do you prefer that?

Comment (Ernst79) [reply src/sensor_state_data/data.py]: Done

**Review by bdraco — APPROVED** [review]

**Pull request merged**